apps/llm-client/src/components/chat/chat-pane.tsx

```
@@ -51,13 +51,38 @@ export function ChatPane() {
+  const [generatingChoices, setGeneratingChoices] = useState(false);
+  const [colorizing, setColorizing] = useState(false);
+  const [choiceRefreshCount, setChoiceRefreshCount] = useState(0);
+
+  // Abort controller for the cancellable color pass
+  const colorAbortRef = useRef<AbortController | null>(null);
+
+  // Cache choices per chat so they survive switching
+  const choiceCacheRef = useRef<Record<string, GeneratedChoice[]>>({});
+  const prevChatIdRef = useRef<string | null>(null);
+
+  useEffect(() => {
```

Biome (2.4.11)

[error] This hook does not specify its dependency on activeChoices.

(lint/correctness/useExhaustiveDependencies)